src/utils/processorUtils.ts

```diff
@@ -0,0 +1,229 @@
+  schema: object
+): ProcessorRequest => {
+  const requestData: ProcessorRequest = { name: formData.name };
+  if (formData.type === ProcessorType.Sink) {
+    requestData.action =
+      isExistingProcessor && formData.action && schema
+        ? {
+            type: formData.action.type,
+            parameters: prepareConfigParameters(
+              formData.action.parameters,
+              schema,
+              "write"
+            ),
+          }
+        : formData.action;
+  } else {
+    requestData.source =
+      isExistingProcessor && formData.source && schema
+        ? {
+            type: formData.source.type,
+            parameters: prepareConfigParameters(
+              formData.source.parameters,
+              schema,
+              "write"
+            ),
+          }
+        : formData.source;
+  }
+  if (formData.filters && formData.filters.length > 0) {
+    const filtersData = prepareFilters(formData.filters) as EventFilter[];
+    if (filtersData.length > 0) {
+      requestData.filters =
+        filtersData as unknown as ProcessorRequest["filters"];
+    }
+  }
+  if (
+    formData.transformationTemplate &&
+    formData.transformationTemplate.trim().length > 0
+  ) {
+    requestData.transformationTemplate = formData.transformationTemplate.trim();
+  }
+  return requestData;
+};
+
+/**
+ * Retrieve the definition of a specific property inside a json schema
+ * @param propertyName Name of the property
+ * @param schema Json schema object
+ * @param [parentSchema] Parent schema object used when drilling down nested fields
+ */
+export const getSchemaDefinition = (
+  propertyName: string,
+  schema: JSONSchema7,
+  parentSchema?: JSONSchema7
+): JSONSchema7Definition | undefined => {
+  if (schema.properties) {
+    const propertySplit = propertyName.split(".");
+    const currentProperty = propertySplit.shift();
+
+    if (currentProperty) {
+      const prop = schema.properties[currentProperty];
+      if (typeof prop === "undefined" || !isJSONSchema(prop)) {
+        return undefined;
+      }
+      const definition: JSONSchema7Definition = prop.$ref
+        ? resolveReference(prop.$ref, parentSchema ?? schema)
+        : prop;
+
+      if (isJSONSchema(definition)) {
+        if (definition.type === "object" && definition.properties) {
+          return getSchemaDefinition(
+            propertySplit.join("."),
+            definition,
+            schema
+          );
+        }
+        return definition;
+      }
+    }
+  }
+  return undefined;
+};
+
+/**
+ * Check if a json schema definition is a json schema and not a boolean
+ * @param value The json schema definition to check
+ */
+export const isJSONSchema = (
+  value: JSONSchema7Definition
+): value is JSONSchema7 => {
+  return typeof value !== "boolean";
+};
+
+/**
+ * Simple reference resolver that works with basic references contained in the
+ * same schema object
+ * @param ref
+ * @param schema
+ */
+export const resolveReference = (
```

Comment:
Really basic reference resolver to retrieve the definition of a property specified in a sub-schema. Only supports basic sub-schemas defined in the same file. (They can be also relative or absolute links to other files)